pkg/ingester/flush.go

```diff
@@ -81,7 +81,16 @@ func (i *Ingester) sweepInstance(instance *instance, immediate bool) {
 }
 
 func (i *Ingester) sweepStream(instance *instance, stream *stream, immediate bool) {
-	if len(stream.chunks) <= 1 && !immediate {
+	old := false
+	if len(stream.chunks) == 1 {
+		chunk := stream.chunks[0]
+		if i.cfg.MaxChunkIdle > 0 &&
+			time.Since(chunk.lastUpdated) > i.cfg.MaxChunkIdle {
+			old = true
+		}
+	}
+
+	if !old && len(stream.chunks) <= 1 && !immediate {
```

Comment:
Code is a bit convoluted - lets flip this on its and turn it into a list of reasons not to flush:

```
if len(stream.chunks)==0 {
  return
}

lastChunk := stream.chunks[0]
if !immediate && time.Since(chunk.lastUpdated) < i.cfg.MaxChunkIdle {
  return
}
```

And I don't think we need do deal with `i.cfg.MaxChunkIdle == 0`, it doesn't make sense to make this optional.